### ISSUE TITLE
`l1loss(gt_img, render_img)` in pytorch only optimize 'base_color' can't optimize `specular` (principled bsdf)

### ISSUE THREAD

**Issue by anewusername77** [mitsuba-renderer/mitsuba3#466]
## Summary

if i use only `l1loss(gt_img, render_img)` during pytorch optimization,  `specular` param can't be updated, unless adding an additional loss for it.

## Steps to reproduce

1. setup scene like in the tutorial, using `principled bsdf` instead
2. pass specular, vertex_positions, base_color of type `torch.Tensor`, whose `requires_grad==True` to render a new img
```python
# optimizer def
optimizer =  torch.optim.Adam([
            {'params': self.vShCoeff, 'lr': 0.007},
            {'params': self.vAlbedoCoeff, 'lr': 0.007},
            {'params': self.vRotation, 'lr': 0.001},
            {'params': self.vTranslation, 'lr': 0.001},
            {'params': self.specular, 'lr': 0.001}
        ])

......
# cam_vertices = torch.matmul( RT, vertices)
# albedo_tex = computeTextureFromAlbedo(computeAlbedo(self.vAlbedoCoeff))
# envmap = computeEnvMapFromSH(self.vShCoeff)
render_img = renderimg(cam_vertices, faces, normals, uvmap, albedo_tex, self.specular, roughness, envmap,fov, 8, it)
......

@dr.wrap_ad(source='torch', target='drjit')
def renderimg(scene, vertices:torch.Tensor, faces:torch.Tensor, normals:torch.Tensor,
              uvmap:torch.Tensor, albedo_tex:torch.Tensor, specular:torch.Tensor, 
              roughness:torch.Tensor, envmap:torch.Tensor, fov, samples, seed):
       # light
        params = mi.traverse(scene)
        params['light.data'] = envmap
        # mesh
        vertex_texcoords = mi.Point2f(mi.Float(uvmap[:,0].numpy()), 
                            mi.Float(uvmap[:,1].numpy()))                                   # how to freeze this parameter?
        params_scene['bfm.vertex_positions'] = dr.ravel(vertices)
        params_scene['bfm.vertex_normals'] = dr.ravel(normals)
        params_scene['bfm.faces'] = dr.ravel(faces)
        params_scene['bfm.vertex_texcoords'] = dr.ravel(vertex_texcoords)
       
        ## bsdf
        params_scene['bfm.bsdf.base_color.data'] = albedo_tex
        params_scene['bfm.bsdf.roughness.value'] = dr.ravel(roughness)
        params_scene['bfm.bsdf.specular'] = dr.ravel(specular)
        
        params_scene.update()
        
        return mi.render(scene, params_scene, seed=seed, spp=samples)
```
4. use l1loss between ground truth img and rendered img.
```python
optimizer.zero_grad()
loss = L1loss(img_gt, rendered_img)
loss.backward()
optimizer.step()
```

in this way, `specular` won't be updated, unless adding an extra loss for it like:`loss += 100 * torch.clamp(-self.specular, min=0).mean()`

**Comment by lynshwoo2022** [comment]
and how to optimize envmap?

**Comment by njroussel** [comment]
This looks related to https://github.com/mitsuba-renderer/drjit/issues/110

We've been accumulating issues with `dr.wrap_ad`. There's definitely something broken in it, hopefully I'll have some time to look at it soon.

**Comment by anewusername77** [comment]
thank you so much. I'll repo if i got any idea.
